great_expectations/rule_based_profiler/parameter_builder/metric_multi_batch_parameter_builder.py

```diff
@@ -118,10 +126,24 @@ def _build_parameters(
         Returns:
             Attributes object, containing computed parameter values and parameter computation details metadata.
         """
+        # Obtain single_batch_mode from "rule state" (i.e., variables and parameters); from instance variable otherwise.
+        single_batch_mode: bool = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=self.single_batch_mode,
+            expected_return_type=bool,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        limit: Optional[int] = None
```

Comment:
You could make this more suctinct if you want: `limit: Optional[int] = 1 if single_batch_mode else None`

Reply:
@billdirks Interesting... The single-batch case is more rare.  Let me look again.

Reply:
You can change the order of the conditional. You are checking `single_batch_mode` in either case.

Reply:
@billdirks I checked -- this is correct, because the `None` limit goes to `DataContext.get_batch_list()` method -- and that is the "no limit" value.  Thanks.

Reply:
I don't understand. The logic either way is the same so both are correct. It's just a matter of succinctness.